src/evm_builtins.rs

```diff
@@ -20,109 +20,99 @@ impl Instructions for EVMInstructions {
         arguments: Vec<SMTVariable>,
         return_vars: &[SMTVariable],
         ssa: &mut SSATracker,
-    ) -> String {
-        let single_return = |value: String| {
+    ) -> SMTStatement {
+        let single_return = |value: SMTExpr| {
             assert_eq!(return_vars.len(), 1);
-            format!(
-                "(define-const {} (_ BitVec 256) {})",
-                &return_vars.first().unwrap().name,
-                value
-            )
+            smt::define_const(return_vars.first().unwrap().clone(), value)
         };
-        let direct = |smt_name: &str| {
-            let smt_encoding =
-                format!("({} {} {})", smt_name, arguments[0].name, arguments[1].name);
-            single_return(match is_bool_function(smt_name) {
+        let direct = |smt_op: SMTOp| {
+            let smt_encoding = SMTExpr {
+                op: smt_op.clone(),
+                args: arguments.clone().into_iter().map(|v| v.into()).collect(),
```

Comment:
```suggestion
                args: arguments.iter().map(|v| v.into()).collect(),
```
?
Once we have the into, it might be enough to juts use `arguments.clone()`.

Reply:
If we do this we need to clone inside the `map`. I'm fine with either, but we do need to clone